app/V1Module/presenters/AssignmentSolutionsPresenter.php

```diff
@@ -264,7 +264,71 @@ public function actionSetBonusPoints(string $id) {
     $this->sendSuccessResponse("OK");
   }
 
-  public function checkSetAcceptedSubmission(string $id) {
+  public function checkSetFlag(string $id) {
+    $solution = $this->assignmentSolutions->findOrThrow($id);
+    if (!$this->assignmentSolutionAcl->canSetFlag($solution)) {
```

Comment:
Just not sure if one canSetFlag will cut it. But maybe we can leave it for now...

Reply:
If you have better solution... I am not really sure if I do...

Reply:
I have, but it is not necessary at the moment.

Reply:
Just because I am curious, what is better solution? If it is explainable in few words?

Reply:
Flag is passed down to check function (since it is in URL), so we can have an ACL policy method for each flag (i.e., canSetFlagAccepted, canSetFlagReviewed). The name of the method can be assembled as a string from flag name (capitalization + prefix) and subsequently called as `$this->assignmentSolutionAcl->$methodName($solution)`.